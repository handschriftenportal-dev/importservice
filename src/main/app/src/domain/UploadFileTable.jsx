import React, {useContext, useEffect, useState} from "react";
import {
  Grid,
  Hidden,
  InputLabel,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TextField,
} from "@material-ui/core";
import {Link} from "react-router-dom";
import {AuthenticationContext} from "../infrastructure/AuthenticationContext";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";

export default function UploadFileTable(props) {

  const [filter, setFilter] = useState('');
  const authentication = useContext(AuthenticationContext)

  useEffect(() => {
    console.log("Rendering UploadFileTabled finished ...")
    return () => {
      console.log("Unmount UploadFileTable");
      setFilter('')
    }
  }, [])

  function onChange(e) {
    setFilter(e.currentTarget.value)
  }

  return (<Grid container justify="center" direction="column"
                alignItems="center">
    <Hidden>
      <Grid item md={1}></Grid>
    </Hidden>
    <Grid container item xs={12} md={10} alignItems="stretch"
          direction="column">
      <Typography component="h3" variant="subtitle1">{'Angemeldeter Benutzer: '
      + authentication.username}</Typography>
      <Paper>
        <Box component="div" visibility="hidden">
          <InputLabel htmlFor="dateinameinput">Dateieingabe</InputLabel>
        </Box>
        <TextField id="dateinameinput" label={'Dateiname filtern'}
                   value={filter}
                   onChange={onChange}></TextField>
        <Table>
          <TableHead>
            <TableRow>
              {props.headlines.map(l => {
                return <TableCell scope="col" key={l}>{l}</TableCell>
              })}
            </TableRow>
          </TableHead>
          <TableBody>
            {
              props.data[0] ? (props.data.filter(
                  job => job.dateiname.toLowerCase().includes(
                      filter.toLowerCase())).map((importJob) => {
                return (
                    <TableRow key={importJob.id} className="uploadfilerow">
                      <TableCell><Link to={'/app/import/job/'
                      + importJob.id}>{importJob.datum}</Link></TableCell>
                      <TableCell>{importJob.benutzer}</TableCell>
                      <TableCell>{importJob.dateianzahl}</TableCell>
                      <TableCell>{importJob.dateiname}</TableCell>
                      <TableCell>{importJob.ergebnis}</TableCell>
                    </TableRow>)
              })) : (<TableRow><TableCell>Keine Daten
                vorhanden</TableCell></TableRow>)
            }
          </TableBody>
        </Table>
      </Paper>
    </Grid>
    <Hidden>
      <Grid item md={1}></Grid>
    </Hidden>
  </Grid>)
}
