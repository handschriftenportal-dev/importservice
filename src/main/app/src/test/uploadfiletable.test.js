import React from "react";
import {unmountComponentAtNode} from "react-dom";
import UploadFileTable from '../domain/UploadFileTable'
import createComponentWithIntl from './createComponentWithIntl';

describe('Import Upload Form Testsuite', () => {

  let container = null;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("Test Upload File Table ", () => {

    const table = createComponentWithIntl(<UploadFileTable data={[{
      "id": "6709b9a0-44a8-4bcd-85ab-0b80a818c5fe",
      "datum": "2020-03-23T16:46:22.627314",
      "benutzer": "Christoph Mackert",
      "dateianzahl": 1,
      "dateiname": "tei-msDesc_Koch.xml",
      "ergebnis": "SUCCESS"
    }]} headlines={['Upload am', 'Benutzer',
      'Anzahl Dateien',
      'Importdatei', 'Ergebnis']}/>, container);

    expect(table.root.findAllByType('table')).toHaveLength(1)
  })

  it("Test Upload File Table Snapshot", () => {

    const table = createComponentWithIntl(<UploadFileTable data={[{
      "id": "6709b9a0-44a8-4bcd-85ab-0b80a818c5fe",
      "datum": "2020-03-23T16:46:22.627314",
      "benutzer": "Christoph Mackert",
      "dateianzahl": 1,
      "dateiname": "tei-msDesc_Koch.xml",
      "ergebnis": "SUCCESS"
    }]} headlines={['Upload am', 'Benutzer',
      'Anzahl Dateien',
      'Importdatei', 'Ergebnis']}/>, container).toJSON();

    expect(table).toMatchSnapshot();
  })
});
