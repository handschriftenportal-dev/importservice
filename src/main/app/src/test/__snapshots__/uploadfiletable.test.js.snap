// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Upload Form Testsuite Test Upload File Table Snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
  >
    <h3
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      Angemeldeter Benutzer: Gast
    </h3>
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiBox-root MuiBox-root-242"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
          htmlFor="dateinameinput"
        >
          Dateieingabe
        </label>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="dateinameinput"
          id="dateinameinput-label"
        >
          Dateiname filtern
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="dateinameinput"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <table
        className="MuiTable-root"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Upload am
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Benutzer
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Anzahl Dateien
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Importdatei
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Ergebnis
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root uploadfilerow"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <a
                href="/app/import/job/6709b9a0-44a8-4bcd-85ab-0b80a818c5fe"
                onClick={[Function]}
              >
                2020-03-23T16:46:22.627314
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Christoph Mackert
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              tei-msDesc_Koch.xml
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              SUCCESS
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
